packages/front-end/components/Features/ScheduleInputs.tsx

```diff
@@ -3,50 +3,77 @@ import clsx from "clsx";
 import { format } from "date-fns";
 import { format as formatTimeZone } from "date-fns-tz";
 import React, { useEffect, useState } from "react";
+import { useUser } from "@/services/UserContext";
 import Field from "../Forms/Field";
 import SelectField from "../Forms/SelectField";
+import PremiumTooltip from "../Marketing/PremiumTooltip";
+import UpgradeMessage from "../Marketing/UpgradeMessage";
+import Toggle from "../Forms/Toggle";
 import styles from "./ScheduleInputs.module.scss";
 
 interface Props {
   defaultValue: ScheduleRule[];
   onChange: (value: ScheduleRule[]) => void;
+  setShowUpgradeModal: (value: boolean) => void;
 }
 
 export default function ScheduleInputs(props: Props) {
   const [rules, setRules] = useState(props.defaultValue);
   const [dateErrors, setDateErrors] = useState({});
+  const { hasCommercialFeature } = useUser();
+  const [toggleEnabled, setToggleEnabled] = useState(() => rules?.length > 0);
```

Comment:
nit: don't really need a function for useState( ()=> ...) since checking length is not expensive